### PR TITLE
SISRP-48130 - Allow APR Link to Remain Active after Students' Last Day of Term for Expected Grad Term

### PR DESCRIPTION
**Pull request by nkutub**
https://jira-secure.berkeley.edu/browse/SISRP-48130
* fix edge case of missing statuses for advisors 
* added more unit tests to cover edge cases

### CODE REVIEW

File: app/models/degree_progress/my_undergrad_requirements.rb [hunk 1]
```diff
@@ -31,15 +31,15 @@ def add_links(response)
 
     def get_incomplete_programs_roles
       ugrd_statuses = MyAcademics::MyAcademicStatus.statuses_by_career_role(@uid, ['ugrd'])
-      return [] if ugrd_statuses.empty?
+      return [] if !ugrd_statuses || ugrd_statuses.empty?
```
Comment (redconfetti) [inline app/models/degree_progress/my_undergrad_requirements.rb]: @nkutub - You could just use `blank?` (the opposite of `present?`)

```ruby
[].present?
 => false
> nil.present?
 => false
> 0.present?
 => true
> {}.present?
 => false

nil.blank?
 => true
> [].blank?
 => true
> {}.blank?
 => true
> 0.blank?
 => false
> -0.blank?
 => false
```
